packages/substrate-codegen/src/dynamic-builder.ts

```diff
@@ -172,5 +172,33 @@ export const getDynamicBuilder = (metadata: V14) => {
     return buildDefinition(storageEntry.type as number)
   }
 
-  return { buildDefinition, buildStorage, buildCall, buildConstant }
+  const buildVariant =
+    (type: "errors" | "events") =>
+    (
+      pallet: string,
+      name: string,
+    ): {
+      codec: Codec<any>
+      location: [number, number]
+    } => {
+      const palletEntry = metadata.pallets.find((x) => x.name === pallet)!
+      const lookup = getLookupEntryDef(palletEntry[type]!)
+      if (lookup.type !== "enum") throw null
```

Comment:
return null or throw null?

Reply:
`palletEntry[type]` is guaranteed to be an `enum` because `type` is either `errors` or `events`. So, this guard is just here to have the correct typings past this line. Meaning that this `throw null` will never happen in runtime, this is just a guard to have the right typings afterwords. Also, keep in mind that this is the **dynamic** builder, meaning that in reality we won't be trying to create this dynamic codec unless the checksums match.

Reply:
ah yeah ur right makes sense